e2e/test/iothub/AuthenticationWithTokenRefreshDisposalTests.cs

```diff
@@ -55,19 +55,77 @@ public async Task DeviceSak_ReusableAuthenticationMethod_SingleDevicePerConnecti
         [LoggedTestMethod]
         public async Task DeviceSak_ReusableAuthenticationMethod_MuxedDevicesPerConnection_Amqp()
         {
-            await ReuseAuthenticationMethod_MuxedDevices(Client.TransportType.Amqp_Tcp_Only, 2);
+            await ReuseAuthenticationMethod_MuxedDevices(Client.TransportType.Amqp_Tcp_Only, 2).ConfigureAwait(false); ;
         }
 
         [LoggedTestMethod]
         public async Task DeviceSak_ReusableAuthenticationMethod_MuxedDevicesPerConnection_AmqpWs()
         {
-            await ReuseAuthenticationMethod_MuxedDevices(Client.TransportType.Amqp_WebSocket_Only, 2);
+            await ReuseAuthenticationMethod_MuxedDevices(Client.TransportType.Amqp_WebSocket_Only, 2).ConfigureAwait(false); ;
+        }
+
+        [LoggedTestMethod]
+        public async Task DeviceClient_AuthenticationMethodDisposesTokenRefresher_Http()
+        {
+            await AuthenticationMethodDisposesTokenRefresher(Client.TransportType.Http1).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task DeviceClient_AuthenticationMethodDisposesTokenRefresher_Amqp()
+        {
+            await AuthenticationMethodDisposesTokenRefresher(Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task DeviceClient_AuthenticationMethodDisposesTokenRefresher_AmqpWs()
+        {
+            await AuthenticationMethodDisposesTokenRefresher(Client.TransportType.Amqp_WebSocket_Only).ConfigureAwait(false);
+        }
+
+        // Even on encountering an exception, the MQTT layer keeps on reattempting CONNECT when communicating via DotNetty's TCP stack.
+        // As a result, instead of throwing the actual exception encountered an IotHubCommunicationException is thrown (on operation timeout).
+        // This is not an issue when the communication is over websockets (where we use the sdk's websocket implementation).
+        // This test has been ignored until we root-cause the issue on DotNetty's TCP stack.
+        [Ignore]
```

Comment:
I just want to call out that mqtt going into endless reconnects isn't a behavior change, this has always been the case.
Previously the sdk would keep on hitting an NRE but still keep retrying. Now it'll hit an ObjectDisposedException instead, but still keep retrying. Based on my investigation this seems to be stemming from the tcp stack implementation used in DotNetty, but I don't have the exact root cause yet.
I am ignoring this test so that we can get this change checked in, while continuing investigation on this issue.